etcdserver/api/v3rpc/maintenance.go

```diff
@@ -45,19 +46,45 @@ type RaftStatusGetter interface {
 	Leader() types.ID
 }
 
+type AuthGetter interface {
+	AuthInfoFromCtx(ctx context.Context) (*auth.AuthInfo, error)
+	AuthStore() auth.AuthStore
+}
+
 type maintenanceServer struct {
```

Comment:
can there be a separate type returned by `NewMaintenanceServer`:
```
type authMaintenanceServer struct {
    ag AuthGetter
    ms *maintenanceServer
}
```
that does the auth checks before calling into the auth-free maintenance server? This buys type-checking that auth covers all calls for the service since if it misses a check it won't match the interface. I'd like to avoid mingling auth code with non-auth code if it's not strictly necessary...